apps/src/code-studio/components/SoundPicker.jsx

```diff
@@ -0,0 +1,115 @@
+import React from 'react';
+import AssetManager from './AssetManager';
+import color from "../../util/color";
+import { SOUND_PREFIX, DEFAULT_SOUND_PATH_PREFIX } from '../../assetManagement/assetPrefix';
+import SoundLibrary from './SoundLibrary';
+
+const audioExtension = '.mp3';
+const styles = {
+  root: {
+    margin: "0 0 0 5px"
+  },
+  divider: {
+    borderColor: color.purple,
+    margin: '5px 0'
+  },
+  warning: {
+    color: color.red,
+    fontSize: 13,
+    fontWeight: 'bold',
+  },
+};
+const MODE = {
+    files : 'files',
+    sounds : 'sounds'
+};
+
+/**
+ * A component for managing hosted sounds and the Sound Library.
+ */
+const SoundPicker = React.createClass({
+  propTypes: {
+    assetChosen: React.PropTypes.func,
+    assetsChanged: React.PropTypes.func,
+    typeFilter: React.PropTypes.string,
+    uploadsEnabled: React.PropTypes.bool.isRequired,
+    showUnderageWarning: React.PropTypes.bool.isRequired,
+    useFilesApi: React.PropTypes.bool.isRequired
+  },
+
+  getInitialState() {
+    return {mode: MODE.files};
+  },
+
+  getAssetNameWithPrefix(sound) {
+    const soundName = sound.replace(DEFAULT_SOUND_PATH_PREFIX, SOUND_PREFIX);
+    this.props.assetChosen(soundName);
+  },
+
+  setSoundMode() {
+    this.setState({mode: MODE.sounds});
+  },
+
+  setFileMode() {
+    this.setState({mode: MODE.files});
+  },
+
+  render() {
+    const isFileMode = this.state.mode === MODE.files;
+    const headerStyles = {
+      fileModeToggle: {
+        float: 'left',
+        margin: '0 20px 0 0',
+        fontFamily: isFileMode ? '"Gotham 5r"' : null,
+        color: isFileMode ? null : '#999',
+        fontSize: 16,
+        cursor: 'pointer'
+      },
+      soundModeToggle: {
+        margin: 0,
+        fontSize: 16,
+        fontFamily: isFileMode ? null : '"Gotham 5r"',
+        color: isFileMode ? '#999' : null,
+        cursor: 'pointer'
+      },
+    };
+
+    let modeSwitch, title = this.props.assetChosen ?
+      <p>Choose Sounds</p> :
+      <p>Manage Sounds</p>;
```

Comment:
nit: This could also be:
```
title = (
  <p>
    {this.props.assetChosen ? "Choose Sounds" : "Manage Sounds"}
  </p>
);
```
Which perhaps makes it slightly clearer that the only thing changing is the next (but also might just be slight personal preference).